tests/bud.bats

```diff
@@ -897,3 +897,58 @@ load helpers
   [ "$status" -eq 0 ]
   [[ $output =~ "/var/file2" ]]
 }
+
+@test "bud with FROM AS construct" {
+  buildah bud --signature-policy ${TESTSDIR}/policy.json -t test1 ${TESTSDIR}/bud/from-as
+  run buildah --debug=false images -a
+  [[ $output =~ "test1" ]]
+  [ "${status}" -eq 0 ]
+
+  ctr=$(buildah --debug=false from --signature-policy ${TESTSDIR}/policy.json test1)
+  run buildah --debug=false containers -a
+  [[ $output =~ "test1" ]]
+  [ "${status}" -eq 0 ]
+  
+  run buildah inspect --format "{{.Docker.ContainerConfig.Env}}" --type image test1
+  echo "$output"
+  [ "$status" -eq 0 ]
+  [[ $output =~ "LOCAL=/1" ]]
+}
+
+@test "bud with FROM AS construct with layers" {
+  buildah bud --layers --signature-policy ${TESTSDIR}/policy.json -t test1 ${TESTSDIR}/bud/from-as
+  run buildah --debug=false images -a
+  [[ $output =~ "test1" ]]
+  [ "${status}" -eq 0 ]
+
+  ctr=$(buildah --debug=false from --signature-policy ${TESTSDIR}/policy.json test1)
+  run buildah --debug=false containers -a
+  [[ $output =~ "test1" ]]
+  [ "${status}" -eq 0 ]
+  
+  run buildah inspect --format "{{.Docker.ContainerConfig.Env}}" --type image test1
+  echo "$output"
+  [ "$status" -eq 0 ]
+  [[ $output =~ "LOCAL=/1" ]]
+}
+
+@test "bud with FROM AS skip FROM construct" {
+  run buildah --debug=false bud --signature-policy ${TESTSDIR}/policy.json -t test1 -f ${TESTSDIR}/bud/from-as/Dockerfile.skip ${TESTSDIR}/bud/from-as
+  [[ $output =~ "LOCAL=/1" ]]
+  [[ $output =~ "LOCAL2=/2" ]]
+  [ "${status}" -eq 0 ]
+
+  run buildah --debug=false images -a
+  [[ $output =~ "test1" ]]
+  [ "${status}" -eq 0 ]
+
+  ctr=$(buildah --debug=false from --signature-policy ${TESTSDIR}/policy.json test1)
+  run buildah --debug=false containers -a
+  [[ $output =~ "test1" ]]
+  [ "${status}" -eq 0 ]
+  
```

Comment:
I think we need to check and ensure that $ctr/1 exists and $ctr/2 does not, somewhere around here, to verify that the third stage's starting point and the final image are based on the first named stage and not altered by anything we do for the second:
```
  run test -e $ctr/1
  [ "${status}" -eq 0 ]
  run test -e $ctr/2
  [ "${status}" -ne 0 ]
```
Either that or having the third stage perform the equivalent check as RUN statements.

Reply:
LGTM once we add this.

Reply:
good thought, will add that.